modules/devguide/examples/go/fle.go

```diff
@@ -0,0 +1,184 @@
+	// The key ID that was written is then used on the decrypt side to find the corresponding key from the store.
+	keyOneEncrypter := provider.EncrypterForKey(key1.ID)
+
+	// We register the providers for both encryption and decryption.
+	// The alias used here is the value which corresponds to the "encrypted" field annotation.
+	err := mgr.RegisterEncrypter("one", keyOneEncrypter)
+	if err != nil {
+		panic(err)
+	}
+
+	err = mgr.RegisterEncrypter("two", provider.EncrypterForKey(key2.ID))
+	if err != nil {
+		panic(err)
+	}
+
+	// We don't need to add a default encryptor but if we do then any fields with an
+	// empty encrypted tag will use this encryptor.
+	err = mgr.DefaultEncrypter(keyOneEncrypter)
+	if err != nil {
+		panic(err)
+	}
+
+	// We only set one decrypter per algorithm.
+	// The crypto manager will work out which decrypter to use based on the `alg` field embedded in the field data.
+	// The decrypter will use the key embedded in the field data to determine which key to fetch from the key store for decryption.
+	err = mgr.RegisterDecrypter(provider.Decrypter())
+	if err != nil {
+		panic(err)
+	}
+	// end::provider[]
+
+	// tag::transcoder[]
+	// Create our transcoder, not setting a base transcoder will cause it to fallback to the
+	// SDK JSON transcoder.
+	transcoder := gocbfieldcrypt.NewTranscoder(nil, mgr)
+
+	// Register the encryption transcoder with the SDK.
+	opts := gocb.ClusterOptions{
+		Authenticator: authenticator,
+		Transcoder:    transcoder,
+	}
+	cluster, err := gocb.Connect("localhost", opts)
+	if err != nil {
+		panic(err)
+	}
+	// end::transcoder[]
+
+	// tag::upsert[]
+	col := cluster.Bucket("default").DefaultCollection()
+
+	person := Person{
+		FirstName: "Barry",
+		LastName:  "Sheen",
+		Password:  "bang!",
+		Addresses: []PersonAddress{
+			{
+				HouseName:  "my house",
+				StreetName: "my street",
+			},
+			{
+				HouseName:  "my other house",
+				StreetName: "my other street",
+			},
+		},
+		Phone: "123456",
+	}
+
+	_, err = col.Upsert("p1", person, nil)
+	if err != nil {
+		panic(err)
+	}
+	// end::upsert[]
+
+	// tag::getmap[]
+	res, err := col.Get("p1", nil)
+	if err != nil {
+		panic(err)
+	}
+
+	var resData map[string]interface{}
+	err = res.Content(&resData)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("%+v\n", resData)
+	// end::getmap[]
+
+	// tag::getstr[]
+	personRes, err := col.Get("p1", nil)
+	if err != nil {
+		panic(err)
+	}
+
+	var personResData Person
+	err = personRes.Content(&personResData)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("%+v\n", personResData)
+	// end::getstr[]
+}
+
+func oldAnnotation() {
+	// tag::oldannotation[]
+	mgr := gocbfieldcrypt.NewDefaultCryptoManager(&gocbfieldcrypt.DefaultCryptoManagerOptions{
+		EncryptedFieldPrefix: "__crypt",
+	})
+	// end::oldannotation[]
+
+	keyring := gocbfieldcrypt.NewInsecureKeyring()
+
+	// tag::legacy[]
+	decrypter := gocbfieldcrypt.NewLegacyAes256CryptoDecrypter(keyring, "mykey", "myhmackey")
```

Comment:
Looks like this is failing the build check, is `NewLegacyAes256CryptoDecrypter` taking the right arguments?

From what I can see in `gocbencryption` it seems to need a `Keyring` and a `LegacyKeyFn`:
https://github.com/couchbase/gocbencryption/blob/fdf80784a7b4abacec34ce0f5a8886378326d614/legacyaesdecrypter.go#L54

Reply:
Good catch. I wrote this a while ago and missed it whilst updating it today.